apps/fortuna/src/keeper.rs

```diff
@@ -837,3 +846,68 @@ pub async fn track_provider(
         })
         .set(end_sequence_number as i64);
 }
+
+// TODO: fields
+#[tracing::instrument(name = "withdraw_fees", skip_all, fields())]
+pub async fn withdraw_fees_wrapper(
+    contract: Arc<SignablePythContract>,
+    provider_address: Address,
+    poll_interval: Duration,
+) {
+    loop {
+        if let Err(e) = withdraw_fees_if_necessary(contract.clone(), provider_address)
+            .in_current_span()
+            .await
+        {
+            tracing::error!("Withdrawing fees. error: {:?}", e);
+        }
+        time::sleep(poll_interval).await;
+    }
+}
+
+/// Withdraws accumulated fees in the contract as needed to maintain the balance of the keeper wallet.
+pub async fn withdraw_fees_if_necessary(
+    contract: Arc<SignablePythContract>,
+    provider_address: Address,
+) -> Result<()> {
+    let provider = contract.provider();
+    let wallet = contract.wallet();
+
+    let keeper_balance = provider
+        .get_balance(wallet.address(), None)
+        .await
+        .map_err(|e| anyhow!("Error while getting balance. error: {:?}", e))?;
+
+    let provider_info = contract
+        .get_provider_info(provider_address)
+        .call()
+        .await
+        .map_err(|e| anyhow!("Error while getting provider info. error: {:?}", e))?;
+
+    let fees = provider_info.accrued_fees_in_wei;
```

Comment:
we should add some check that the fee_manager is actually the keeper. Otherwise, we may get confusing revert errors